sdk/api.js

```diff
@@ -2,175 +2,252 @@
-                } else if(options.chunk.combine == 'balances') {
-                  output.push(call.output);
-                }
-              }
-              resolve();
-            } catch(error) {
-              reject(error);
+            if (process.env.LOG_PROGRESS == 'true') {
+              progressBar.update(complete / chunks.length);
             }
-          });
-        }
 
-        const limiter = new Bottleneck({
-          maxConcurrent: process.env.ADAPTER_CONCURRENCY || 1
-        });
+            if (call.ethCallCount) {
+              ethCallCount += call.ethCallCount;
 
-        await Promise.all(_.map(chunks, (chunk) => {
-          return limiter.schedule(() => {
-            return processRequest(chunk);
-          });
-        }));
+              if (options.chunk.combine == 'array') {
+                output = [
+                  ...output,
+                  ...call.output
+                ]
+              } else if (options.chunk.combine == 'balances') {
+                output.push(call.output);
+              }
+            }
+            resolve();
+          } catch (error) {
+            reject(error);
+          }
+        });
+      }
 
-        if(process.env.LOG_PROGRESS == 'true') {
-          progressBar.update(1);
-        }
+      const limiter = new Bottleneck({
+        maxConcurrent: process.env.ADAPTER_CONCURRENCY || 1
+      });
 
-        if(options.chunk.combine == 'balances') {
-          console.log('balances combine')
-          output = utility.sum(output);
-        }
+      await Promise.all(_.map(chunks, (chunk) => {
+        return limiter.schedule(() => {
+          return processRequest(chunk);
+        });
+      }));
 
-        return {
-          ethCallCount,
-          output
-        }
-      } else {
-        let url = `${process.env.DEFIPULSE_API_URL}/${process.env.DEFIPULSE_KEY}${endpoint}`;
+      if (process.env.LOG_PROGRESS == 'true') {
+        progressBar.update(1);
+      }
 
-        if(process.env.INFURA_KEY) {
-          url = `${url}?infura-key=${process.env.INFURA_KEY}`;
-        }
+      if (options.chunk.combine == 'balances') {
+        console.log('balances combine')
+        output = utility.sum(output);
+      }
 
-        let response = await axios.post(url, options);
+      return {
+        ethCallCount,
+        output
+      }
+    } else {
+      let url = `${process.env.DEFIPULSE_API_URL}/${process.env.DEFIPULSE_KEY}${endpoint}`;
 
-        return response.data;
+      if (process.env.INFURA_KEY) {
+        url = `${url}?infura-key=${process.env.INFURA_KEY}`;
       }
-    } catch(error) {
-      throw error.response ? error.response.data : error;
+
+      let response = await axios.post(url, options);
+
+      return response.data;
     }
+    // else if (!ethReq) {
+    //   let client = new kavaSDK.KavaClient(Kava.KavaClient.);
+    //   supply = await client.getSupply()
+    //   // TODO: request or consider other POSTCosmos method
+    // }
```

Comment:
This is where we'll be plugging in the Kava/Cosmos-chain facing request, correct?

Reply:
yeah, although since Kava provides already a wrapper for the client, we don't really use endpoints. I was thinking in deleting this and just using the KavaClient methods. wdyt?

Reply:
Yes, I think plugging in Kava's JS Client here is the cleanest approach as it'll handle sending queries (and it'll be easy to support sending transactions in any future version updates).